### PR TITLE
WEB-2442 auto scroll table when dragging a column to reorder it and near viewport edge

### PR DESCRIPTION
**Pull request by nburwell**
@garvank here is my PR for auto scroll of table when moving columns

I am going to see if I can clean up the logic as I put in support for both types of tables (overflow auto on the wrapper), versus a more traditional table that spills out past the viewport.  Supporting both made it get a bit messier, plus is coupled right now to a wrapper class we use by convention which I don't like.

### CODE REVIEW

File: spec/plugins/data_table/local.js [hunk 1]
```diff
@@ -294,10 +294,12 @@ describe("DataTable Plugin", function() {
         // we need to test this using an async strategy because the checkbox is toggled async as well
         table.dataTable.on("page", function() {
           _.defer(function() {
-            table.page("previous");
             expect(table.$("th.bulk :checkbox").prop("checked")).toEqual(true);
+            done();
```
Comment (nburwell) [inline spec/plugins/data_table/local.js]: this test was not correct before, and was throwing warnings in `grunt spec`, now it's fixed

**Comment by nburwell** [comment]
@garvank I removed the coupling to our wrapper class and got end to end tests for both overflow auto mode, and traditional browser scrollbar mode

File: examples_server/examples/data_table/large_table_column_reordering/data_table.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+Backdraft.app("TableExample", {
+
+  plugins : [ "DataTable"],
```
Comment (garvank) [inline examples_server/examples/data_table/large_table_column_reordering/data_table.js]: Styleguide calls for no spaces between key name and `:`, looks like all keys in this file have a single space.

File: spec/plugins/data_table/general.js [hunk 3]
```diff
@@ -814,6 +814,177 @@ describe("DataTable Plugin", function() {
       });
     });
 
+    describe("autoscroll on dragging of columns", function() {
+      beforeEach(function() {
+        app.view.dataTable.row("R", {
+          columns : [
```
Comment (garvank) [inline spec/plugins/data_table/general.js]: More spaces after key names here

File: examples_server/examples/data_table/large_table_column_reordering/data_table.js (hunk 2 above)
Comment (nburwell) [inline examples_server/examples/data_table/large_table_column_reordering/data_table.js]: Good point, I copied this from the other examples.. which were all made back before the styleguide.  I'll see at making a pass across all the examples in backdraft!

**Pull request merged**